### PR TITLE
Refactor model loading

### PR DESCRIPTION
**Pull request by ZzzhHe**
## Description: 

+  Reorganized the model structure by refactoring the **Vertex**, **Face**, and **Model** classes for improved modularity and maintainability.
+ Replaced the **`tinyobjloader`** with  the [**`assimp`**](https://github.com/assimp/assimp) to support a wide range of model format in the future. ( currently the project still focus on loading the `.obj` file.)

## Test Platform
macOS 14.5
Linux (Ubuntu 20.04)

## TODO:
+ Optimize the code for drawing lines and triangles to improve performance.
+ refactor the `simple_renderer` class to split it into several smaller, more focused classes.
+ Refactor the data buffer to support additional information.

Screenshot 2024-08-08 at 17 40 46

### CODE REVIEW

**Comment by MRNIU** [comment]
这笔 pr 与 #11 似乎有部分重复了，我们一笔一笔合吧

**Comment by ZzzhHe** [comment]
根据 #11 的review重新更改了这笔，已通过build部分的CI

File: src/include/config.h
Comment (MRNIU) [inline src/include/config.h:1]: 这个是生成的文件，需要 ignore 掉

File: src/include/material.hpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#pragma once
```
Comment (MRNIU) [inline src/include/material.hpp]: 头文件的 gurd 风格需要统一

**Pull request merged**